### PR TITLE
Tag smoke test images with date and run ID from workflow.

### PR DESCRIPTION
**Pull request by anuraaga**
I think it's too confusing to have three different tags (based on the seconds value) for the same image but different JRE. We have a perfectly good identifier that's consistent and back-referencable, the github sha :)

Changelog:

Tooling:
- Smoke test images now are tagged with a consistent tag per build of "DATE.TIME.GITHUB_RUN_ID"

### CODE REVIEW

**Review by trask — APPROVED** [review]

**Comment by iNikem** [comment]
sha does not have order (older-younger). Stable tags are good, sha is bad :)

**Comment by vovencij** [comment]
I would suggest to use some kind of stable run number from github action (which doesn't change when action is re-run on error). And reduce the tag to `yyyyMMdd.<buildNumber>` then we will not produce new images on re-running the action.

File: .github/workflows/build-grpc-smoke-dist.yaml [hunk 1]
```diff
@@ -28,7 +28,9 @@ jobs:
 
       - name: Build Docker Image
         run: |
-          ./gradlew jib -PtargetJDK=8 -Djib.httpTimeout=120000 -Djib.console=plain --stacktrace
-          ./gradlew jib -PtargetJDK=11 -Djib.httpTimeout=120000 -Djib.console=plain --stacktrace
-          ./gradlew jib -PtargetJDK=15 -Djib.httpTimeout=120000 -Djib.console=plain --stacktrace
+          TAG=$(date '+%Y%m%d.%H%M%S')
```
Comment (iNikem) [inline .github/workflows/build-grpc-smoke-dist.yaml:31]: Please think about @vovencij suggestion about using date+number as more stable tag.

**Comment by anuraaga** [comment]
Thanks - used the format.

**Comment by anuraaga** [comment]
Oops - didn't notice the removal of seconds but sounds nice!

**Comment by anuraaga** [comment]
Confirmed logic here - https://github.com/anuraaga/opentelemetry-java-instrumentation/runs/1357364079?check_suite_focus=true

**Review by iNikem — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Pull request merged**